### PR TITLE
XWIKI-18630: Replace the AdminWikisTemplatesSheet live table with live data

### PR DESCRIPTION
**Pull request by manuelleduc**

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/src/main/resources/WikiManager/AdminWikisTemplatesSheet.xml [hunk 1]
```diff
@@ -41,25 +41,32 @@
...
+      { 'id': 'membershipType', 'sortable': false, 'filterable': false }
+    ]
+  }
+})
+## TODO: onlyTemplates=1'?
+#set ($sourceParameters = $escapetool.url({
+  'className': 'XWiki.XWikiServerClass',
+  'resultPage': 'WikiManager.WikisLiveTableResults',
+  'translationPrefix': 'platform.wiki.browse.',
+  'onlyTemplates': '1'
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/src/main/resources/WikiManager/AdminWikisTemplatesSheet.xml:61]: @mflorea all the rest is good but I didn't find a way to get rid of this source parameter.

File: xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-test/xwiki-platform-wiki-test-docker/src/test/it/org/xwiki/wiki/test/ui/WikiTemplateIT.java [hunk 2]
```diff
@@ -150,6 +153,14 @@ private void createTemplateWiki() throws Exception
         }
         assertTrue(wikiLink.getURL().endsWith("/xwiki/wiki/mynewtemplate/view/Main/"));
 
+        // Verify the wiki template is displayed the admin wiki templates list.
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-test/xwiki-platform-wiki-test-docker/src/test/it/org/xwiki/wiki/test/ui/WikiTemplateIT.java:156]:
```suggestion
        // Verify the wiki template is displayed in the admin wiki templates list.
```

File: xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/pom.xml [hunk 3]
```diff
@@ -137,6 +137,25 @@
...
+      <artifactId>xwiki-platform-livedata-livetable</artifactId>
+      <version>${project.version}</version>
+      <scope>runtime</scope>
+    </dependency>
+<!--    <dependency>-->
+<!--      <groupId>org.xwiki.platform</groupId>-->
+<!--      <artifactId>xwiki-platform-localization-rest-default</artifactId>-->
+<!--      <version>${project.version}</version>-->
+<!--      <scope>runtime</scope>-->
+<!--    </dependency>-->
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/pom.xml:158]: will be removed

File: xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-test/xwiki-platform-wiki-test-pageobjects/src/main/java/org/xwiki/wiki/test/po/AdminWikisTemplatesSheetPage.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.test.ui.po.ViewPage;
+
+/**
+ * Provides the operations to interact with the Wiki Templates administration page.
+ *
+ * @version $Id$
+ * @since 13.4RC1
+ */
+public class AdminWikisTemplatesSheetPage extends ViewPage
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-test/xwiki-platform-wiki-test-pageobjects/src/main/java/org/xwiki/wiki/test/po/AdminWikisTemplatesSheetPage.java:32]: Why "Sheet"? The fact that there is a sheet used to display the administration section is a technical detail. The PO should reflect what the user sees and interacts with. And from the JavaDoc I understand that the user interacts with the administration section.

File: xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-test/xwiki-platform-wiki-test-pageobjects/src/main/java/org/xwiki/wiki/test/po/AdminWikisTemplatesSheetPage.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+{
+    /**
+     * Goes to the Wiki Template administration page sheet.
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-test/xwiki-platform-wiki-test-pageobjects/src/main/java/org/xwiki/wiki/test/po/AdminWikisTemplatesSheetPage.java:35]: Why go to the sheet? The sheet is a technical page which is hidden from the user. I doubt the administrators ever access the sheet page. They normally access the corresponding wiki administration section. So this PO should extend an administration section no?

File: xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/pom.xml (hunk 3 above)
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/pom.xml:158]: This needs to be removed.

File: xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/src/main/resources/WikiManager/AdminWikisTemplatesSheet.xml (hunk 1 above)
Comment (mflorea) [reply xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/src/main/resources/WikiManager/AdminWikisTemplatesSheet.xml:61]: That's fine, it's a custom source parameter. See for instance https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-livetable/src/test/resources/liveTableConfigHelper/NotificationsFiltersPreferencesMacro.test#L22 . The items from ``extraParams`` that cannot be converted to a query (property) filter or to some other standard live data parameter must be kept as custom source parameters.

**Review by mflorea — CHANGES_REQUESTED** [review]

File: xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/src/main/resources/WikiManager/AdminWikisTemplatesSheet.xml (hunk 1 above)
Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/src/main/resources/WikiManager/AdminWikisTemplatesSheet.xml:61]: ok, thanks for confirming, cleaning up the PR and merging then.

**Pull request merged**